Add commands to interactive mode.

wbpxre150 opened the pull request:
I thought of this while working on something else. 
Its super useful. You can change startup params on the fly using the command: 
`??? CMD param`
where CMD is a command line argument, and params is the number following each argument. 

In my testing so far changing values has changed the models output. 

This should enable users to test different params in run time against a single prompt. We can now fine tune the params using a single prompt. Simply copy and paste the prompt over and over after changing the param and compare the output.

wbpxre150 committed:
add exit call to interactive mode.

wbpxre150 committed:
move code to signal handler

wbpxre150 committed:
add command line to interactive mode. You can specify diffrent values for everything in params during rumtime.

wbpxre150 committed:
fixes

wbpxre150 committed:
fixO

wbpxre150 committed:
refector code into function.

j-f1 commented on examples/main/main.cpp:
```diff
@@ -47,6 +47,53 @@ void sigint_handler(int signo) {
 
+void command(std::string buffer, gpt_params params, const int n_ctx ) {
+    // check buffer's first 3 chars equal '???' to enter command mode. 
+    if (strncmp(buffer.c_str(), "???", 3) == 0 && buffer.length() > 3) {
```
Nit: please negate this condition and do an easy return so the function body doesn't have to be indented again:
```suggestion
    if (buffer.length() < 3 || strncmp(buffer.c_str(), "???", 3) != 0) return;
```

wbpxre150 replied:
done mate.

wbpxre150 committed:
fix code indentation.

wbpxre150 committed:
Fix prints.

wbpxre150 committed:
whitespace

wbpxre150 committed:
fix conflict

wbpxre150 commented:
It would print random gibberish after entering a command. This is now fixed. I also cleaned up the code and changed a few prints to make more sense. This should be ready for review now.

wbpxre150 committed:
Merge pull request #1 from ggerganov/master

pull down new code

wbpxre150 committed:
Merge pull request #2 from wbpxre150/wbpxre150

merge command line code

wbpxre150 committed:
Merge pull request #3 from wbpxre150/master

merge new code

wbpxre150 committed:
Merge branch 'master' into wbpxre150
